src/ec2/tools.ts

```diff
@@ -0,0 +1,36 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { VscodeRemoteSshConfig } from '../shared/extensions/ssh'
+import { Result } from '../shared/utilities/result'
+
+export class Ec2RemoteSshConfig extends VscodeRemoteSshConfig {
+    private readonly command: string =
+        "aws ssm start-session --target %h --document-name AWS-StartSSHSession --parameters 'portNumber=%p'"
+    protected override proxyCommandRegExp = new RegExp(`/proxycommand.{0,1024}${this.command}.{0,99}/i`)
+
+    public override async ensureValid() {
+        const proxyCommand = await this.getProxyCommand(this.command)
+        if (proxyCommand.isErr()) {
+            return proxyCommand
+        }
+
+        const verifyHost = await this.verifySSHHost(proxyCommand.unwrap())
+        if (verifyHost.isErr()) {
+            return verifyHost
+        }
+
+        return Result.ok()
+    }
+
+    protected override createSSHConfigSection(proxyCommand: string): string {
+        return `
+# Created by AWS Toolkit for VSCode. https://github.com/aws/aws-toolkit-vscode
+Host i-* mi-*
```

Comment:
- let's use `aws-ec2-` as the prefix, for now.
- only need one value for the hostname. the only purpose of the hostname is to "route" to our ProxyCommand.

```suggestion
Host aws-ec2-*
```